sdk/core/core/src/az_http_request.c

```diff
@@ -188,8 +188,17 @@ AZ_NODISCARD az_result az_http_request_get_parts(
     az_span* out_url,
     az_span* out_body)
 {
-  *out_method = request->_internal.method;
-  *out_url = az_span_slice(request->_internal.url, 0, request->_internal.url_length);
-  *out_body = request->_internal.body;
+  if (out_method != NULL)
```

Comment:
Why are we allowing null ptrs here rather than these being preconditions? At the very least this seems unrelated to the fix itself.

Reply:
not for the issue, right
This is part of a general problem where transport adapter is accesing internal fields.
I can remove this from this PR and add it later.
This is to allow calling get parts just to get method with out worrying about headers and body from a request

Reply:
Oh I see. So is this so that the caller can pick which part they are interested in and pass in `NULL` for the parts they don't need?

Reply:
Please update the docs for `az_http_request_get_parts` to mention this.